### ISSUE TITLE
Generate classes from readonly fields

### ISSUE THREAD

**Issue by bboyle1234** [amis92/RecordGenerator#103]
Sometimes an immutable class should have readonly fields that are not exposed by a getter property. This allows a lot more flexibility and advanced usage.

```csharp
// The class
[Record(Features.Default | Features.GenerateGetterProperties)]
public sealed partial class SomeObject { 
  private readonly int property1;
  [SuppressGetter] private readonly int property2;
}

// Example code gen
partial class SomeObject { 
  public int Property1 => property1;
  public SomeObject(int property1, int property2) { 
    this.property1 = property1;
    this.property2 = property2;
  }
}
```

**Comment by amis92** [comment]
That's more of a primary constructor feature (https://github.com/dotnet/csharplang/issues/2691). Here we focus on records - always being a public, ordered list of immutable properties. There are variants to be considered (like dataset, which would be an unordered bag of properties), but hiding a ctor parameter doesn't fit in, I think.

**Comment by bboyle1234** [comment]
ok. Thanks for considering. As I commented earlier, the reason for suggesting is the hope of using this library not only to build records, but to build what we really want ... powerful, flexible immutable classes with customizable memberwise equality. Sometimes this flexibility requires that the immutable class be built from fields instead of public readonly properties. 
AArnott demonstrates this well in his `ImmutableObjectGraph` project.
When I first saw the "build from readonly fields" method used there, I didn't like it. Then I thought it over for a few days and see many good reasons to go that way.
